### ISSUE TITLE
[Bug] F-Droid can't build

### ISSUE THREAD

**Issue by licaon-kter** [w2sv/FileNavigator#32]
ref: https://monitor.f-droid.org/builds/log/com.w2sv.filenavigator/5#site-footer

so looks like you tagged https://github.com/w2sv/FileNavigator/releases/tag/0.1.0 from https://github.com/w2sv/FileNavigator/commit/d1602474ec5f5f3439777f1e8384a9eac75b0079

but the APK was built from 3ffc571250e85f39cce517c3d8d337083f3c85c8

yet if I build from that
```
==== detail begin ====
verification of APK with copied signature failed
Comparing reference APK to APK with copied signature...
Unexpected diff output:
Binary files /tmp/tmp1hkqq9hy/unsigned_binaries_com.w2sv.filenavigator_5.binary/content/assets/dexopt/baseline.prof and /tmp/tmp1hkqq9hy/_tmp_tmp1hkqq9hy_sigcp_com.w2sv.filenavigator_5/content/assets/dexopt/baseline.prof differ
Binary files /tmp/tmp1hkqq9hy/unsigned_binaries_com.w2sv.filenavigator_5.binary/content/classes.dex and /tmp/tmp1hkqq9hy/_tmp_tmp1hkqq9hy_sigcp_com.w2sv.filenavigator_5/content/classes.dex differ
==== detail end ====
```

/LE: fyi https://gitlab.com/fdroid/fdroiddata/-/commit/e83b9f17de7b787ce3877c07794e8014012e9483

**Comment by w2sv** [comment]
I think it was the other way around, I built the APK from [d160247](https://github.com/w2sv/FileNavigator/commit/d1602474ec5f5f3439777f1e8384a9eac75b0079), then my Make publishing routine threw an error during the upload to the playstore, because the release notes were too long, so I changed them and thereupon tagged the release from [3ffc571](https://github.com/w2sv/FileNavigator/commit/3ffc571250e85f39cce517c3d8d337083f3c85c8). I don't know how that could possibly impact the generated baseline profile or the dex classes, though.

**Comment by w2sv** [comment]
I could simply create a new release, I have one at the ready anyways. Would that be a valid workaround? :D

**Comment by licaon-kter** [comment]
Please do

**Comment by w2sv** [comment]
Done: https://github.com/w2sv/FileNavigator/releases/tag/0.1.1

**Comment by licaon-kter** [comment]
you tagged https://github.com/w2sv/FileNavigator/releases/tag/0.1.1 from 8ee0e8060c30075c4cefb66a16330c6594fa0758

but you built from 6f5a0ab3fafb0979f39a2e26ae00a1606cc2e799

```
diff -r /tmp/tmprxk9qcuj/unsigned_binaries_com.w2sv.filenavigator_6.binary/content/META-INF/version-control-info.textproto /tmp/tmprxk9qcuj/_tmp_tmprxk9qcuj_sigcp_com.w2sv.filenavigator_6/content/META-INF/version-control-info.textproto
4c4
<   revision: "6f5a0ab3fafb0979f39a2e26ae00a1606cc2e799"
---
>   revision: "8ee0e8060c30075c4cefb66a16330c6594fa0758"
```

**Comment by w2sv** [comment]
Uff...

**Comment by w2sv** [comment]
Next try I guess: https://github.com/w2sv/FileNavigator/releases/tag/0.1.2

Sorry...

**Comment by licaon-kter** [comment]
thanks https://gitlab.com/fdroid/fdroiddata/-/commit/f6dac050273af7328b94b34a5fd0febb96602c27

**Comment by w2sv** [comment]
Thank you.

**Comment by licaon-kter** [comment]
you tagged https://github.com/w2sv/FileNavigator/releases/tag/0.3.0 from https://github.com/w2sv/FileNavigator/commit/32b9e610931c343e8a9bf719b5c48de71df3cefd

but the APK was built from a newer commit c7552b4d267c5b3df2b9765a6e8e54dd4e12571c ? I see this is https://gitlab.com/fdroid/checkupdates-bot-fdroiddata/-/jobs/10001578356#L637

now, I update the recipe to that commit but still fails... :( https://gitlab.com/fdroid/fdroiddata/-/jobs/10001989501

**Comment by w2sv** [comment]
Yeah not sure what happened there. I created the release at the end of a long day full of programming, I wasn't at my mentally freshest anymore, sorry 🙃

I guess I'll just create a new release, that should probably fix it. Do I need to increment the version/versionCode for that?

**Comment by licaon-kter** [comment]
that's up to you

**Comment by licaon-kter** [comment]
difflog: nope :(

[com.w2sv.filenavigator_15.log](https://github.com/user-attachments/files/20164495/com.w2sv.filenavigator_15.log)

**Comment by IzzySoft** [comment]
Though release 0.3.0.1 now is already distributed, so this would need a fresh release to reach everyone (and yes, reproducible build failed at IzzyOnDroid as well, diffs suggest a "dirty build" as @linsui just pointed out (so make sure to run `clean` before your next release build).

**Comment by IzzySoft** [comment]
> which seems to have 20 cores and 28 threads

Urn… Afraid we cannot match that on a machine with 16 cores…

> I'm indeed not building/releasing via a Github action, but I can certainly set up one to do that.

That'd be great! That environment we mostly can match.

**Comment by w2sv** [comment]
Alright I got it: https://github.com/w2sv/FileNavigator/releases/tag/0.3.1.1
Fingers crossed for it to work 🤞

**Comment by licaon-kter** [comment]
the APK in https://github.com/w2sv/FileNavigator/releases/tag/0.3.1.1appears unsigned https://github.com/w2sv/FileNavigator/releases/tag/0.3.1.1 🤷

**Comment by IzzySoft** [comment]
@w2sv ^^ we'll demote your app to monthly update checks now, else the APK is pulled in each day just to be thrown away again. Might have to disable updates altogether if you cannot fix it 🙈

**Comment by IzzySoft** [comment]
Err… 0.3.1? We already have that, it was released 2025-06-23. Then there was even a 0.3.1.1 (which used the same `versionCode`). And now, you go back and have another 0.3.1? A bit confusing. I mean, I can try to manually force it in – but 2 versions with the same name would … well, look not quite right ("what's the difference between 0.3.1 and 0.3.1-fdroid? Why are there both?"). Also,

> `Full Changelog: 0.3.1.1...0.3.1-fdroid`

looks a bit like a downgrade 🙈 Why not 0.3.2, or 0.3.1.2? Also see: [semantic versioning](https://semver.org/) 😉

**Comment by w2sv** [comment]
Because it aims exclusively at fixing the fdroid build and does not contain any source code changes or dependency updates compared to 0.3.1 at all. Same applied for 0.3.1.1, but I didn't like that. Incrementing the version only makes sense if there were changes to the software, but I did it anyways then just to trigger a new build. That's why I now used the fdroid suffix instead - also because I use it as a GitHub action trigger for the fdroid-only release.
Moving forward I'll drop the fdroid suffix again on "normal" releases, that actually contain changes to the app.
If it produces overhead to accommodate this new suffix there's no need to bother, I'll release an actual new app version soon.

**Comment by linsui** [comment]
Please just upload a new apk. :)
